queries.sql

```diff
@@ -137,4 +137,77 @@ FROM owners AS o
 JOIN animals AS a ON a.owners_id = o.id
 GROUP BY o.full_name
 ORDER BY 2 DESC
-LIMIT 1;
\ No newline at end of file
+LIMIT 1;
+
+-- Write queries to answer the following:
+-- Who was the last animal seen by William Tatcher?
+SELECT a.name, MIN(vi.visit_date) 
+FROM visits AS vi
+LEFT JOIN vets AS ve ON vi.vet_id = ve.id
+LEFT JOIN animals AS a ON ve.id = a.id
+WHERE ve.name = 'William Tatcher'
+GROUP BY a.name;
+-- How many different animals did Stephanie Mendez see?
+SELECT ve.name, COUNT(*) AS qty_animals_visited
+FROM visits AS vi
+LEFT JOIN vets AS ve ON vi.vet_id = ve.id
+WHERE ve.name = 'Stephanie Mendez'
+GROUP BY ve.name;
+-- List all vets and their specialties, including vets with no specialties.
+SELECT v.name AS vet_name, s.name AS specialization
+FROM vets AS v
+LEFT JOIN specializations AS sp ON v.id = sp.vet_id
+LEFT JOIN species AS s ON sp.species_id = s.id
+ORDER BY v.name, s.name;
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT ve.name, vi.visit_date
+FROM visits AS vi
+LEFT JOIN vets AS ve ON vi.vet_id = ve.id
+WHERE ve.name = 'Stephanie Mendez' 
+AND vi.visit_date BETWEEN '2020-04-1' AND '2020-08-30'
+ORDER BY ve.name, vi.visit_date
+-- What animal has the most visits to vets?
+SELECT a.name, MAX(vi.vet_id) 
+FROM visits AS vi
+LEFT JOIN vets AS ve ON vi.vet_id = ve.id
+LEFT JOIN animals AS a ON ve.id = a.id
+GROUP BY a.name
+LIMIT 1;
```

Comment:
- [ ] Since we are looking for the animals that have the most visits to vets, I think the best way is to count the number of visits that each animal has and order them in descendent order to finally pick the first. With that in mind, the code could look like this :point_down: 

```sql
  SELECT animals.name, count(animals.name)
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  ORDER BY count(animals.name) DESC
```